CRM/Core/BAO/CustomGroup.php

```diff
@@ -2259,7 +2259,7 @@ public static function getSubTypes(): array {
     $sel2['Campaign'] = $campaignTypes;
     $sel2['Membership'] = $membershipType;
     $sel2['ParticipantRole'] = $participantRole;
-    $sel2['ParticipantEventName'] = CRM_Event_PseudoConstant::event(NULL, FALSE, "( is_template IS NULL OR is_template != 1 )");
+    $sel2['ParticipantEventName'] = CRM_Event_PseudoConstant::event(NULL, FALSE, "( is_template != 1 )");
```

Comment:
This. This right here is the reason we need to fix our schema. No more ugly workarounds for unreliable boolean fields!

Reply:
I wonder if there's an argument to treat these kind of schema changes the same as deprecations, where there's a period of time before actually changing the code that uses the old value. Like `// @deprecated can remove check for null in 5.55`.

I don't have a specific instance for this code line, but there's been issues before where upgrade code in an intermediate version between the initial and final version will call some core code such as this indirectly, but the upgrade step that this depends on hasn't run yet. E.g.
* Upgrading from 5.47 to 5.49
* Pretend that 5.48 had some upgrade code that hits this line.
* At the point that upgrade code runs, is_template could still be null in the db, but the line has the new code.
* Here it won't crash, but might not work as intended.

Reply:
@demeritcowboy I think that's not a bad idea, but this particular field has already been fixed once (just not completely - the upgrade script forgot to add `NOT NULL`) and this code only runs on the form AFAIK.

https://github.com/civicrm/civicrm-core/commit/99f13d8d83f1786a6bd8e2238d0424663488a988#diff-77e9eaeaf4c5c2129efa23ff414bf1657535b3f6e50e66324457f17095c666a6R254-R256